### ISSUE TITLE
🚨 Potential Improperly Controlled Modification of Object Prototype Attributes ('Prototype Pollution') (CWE-1321)

### ISSUE THREAD

**Issue by huntr-helper** [jaredhanson/utils-merge#8]
👋 Hello, @jaredhanson - a potential high severity Improperly Controlled Modification of Object Prototype Attributes ('Prototype Pollution') (CWE-1321) vulnerability in your repository has been disclosed to us.

#### Next Steps

1️⃣  Visit **https://huntr.dev/bounties/1-other-jaredhanson/utils-merge** for more advisory information.

2️⃣  **[Sign-up](https://huntr.dev/)** to validate or speak to the researcher for more assistance.

3️⃣ Propose a patch or outsource it to our community - whoever fixes it gets paid.

---

#### Confused or need more help?

- Join us on our **[Discord](https://huntr.dev/discord)** and a member of our team will be happy to help! 🤗

- Speak to a member of our team: @JamieSlome

---

*This issue was automatically generated by [huntr.dev](https://huntr.dev) - a bug bounty board for securing open source code.*

**Comment by jayateertha043** [comment]
Hi ,
I also published a fix for this at https://github.com/jaredhanson/utils-merge/pull/9
It seems to have passed all the tests, you may merge it if it helped.

Thanks

**Comment by rockarts** [comment]
This passes the tests but your fix doesn't work. It's also just copying #7 and trying to claim a bounty. It's debatable that this is even an issue since most repos rely on the merge functionality that both our issues "fix".

**Comment by snoopysecurity** [comment]
I don' think there is prototype pollution here:

```
var merge = require('utils-merge');
var a = { foo: 'bar', qux: 'corge' }
      , b = {
        "__proto__": {
          "polluted": "true",
        }
      };
    var o = merge(a, b);
    console.log(o.polluted) //o is polluted but not the global object prototype
    console.log({}.polluted) //if there was prototype pollution, this would be true
```

You are polluting your own object, not the global object

**Comment by jayateertha043** [comment]
Yes, I agree this isn't a issue as global object isn't polluted, sorry my bad didn't check that properly. 

You may close the issue.

**Comment by JamieSlome** [comment]
I have appropriately marked the advisory as invalid.

Cheers! 🍰
